.github/workflows/build-docs.yml

```diff
@@ -8,36 +8,142 @@ on:
   # Allows you to run this workflow manually from the Actions tab
   workflow_dispatch:
 
-# Sets permissions of the GITHUB_TOKEN to allow pushing to gh-pages branch
+  # Reusable workflow for PR previews
+  workflow_call:
+    inputs:
```

Comment:
IIRC we also need to define matching inputs under the `workflow_dispatch` event trigger, otherwise the github Actions UI launch for this workflow won't work.